File: BaseSDK/src/test/java/com/kaibo/base/ProgressListener.java
package com.kaibo.base;

/**
 * @author Administrator
 * @date 2017/11/30 0030 14:07
 * @description
 */
public interface ProgressListener {
    /**
     * 上传进度回调
     *
     * @param progress 当前进度    范围  0-1
     */
    void onProgress(float progress);
}

File: Core/src/main/java/com/kaibo/core/http/progress/ProgressListener.kt
package com.kaibo.core.http.progress

import java.util.concurrent.ConcurrentHashMap

/**
 * @author Administrator
 * @date 2018/4/3 0003 上午 9:34
 * GitHub：
 * email：
 * description：
 */

object ProgressListener{
    /**
     * 监听下载进度
     */
    val downloadProgressListeners by lazy {
        ConcurrentHashMap<String, (currentLength: Long, fillLength: Long, done: Boolean) -> Unit>()
    }

    /**
     * 监听上传进度
     */
    val uploadProgressListener by lazy {
        ConcurrentHashMap<String, (currentLength: Long, fillLength: Long, done: Boolean) -> Unit>()
    }
}


